File: app/src/main/java/org/dhis2afgamis/usescases/about/AboutModule.java
package org.dhis2afgamis.usescases.about;

import androidx.annotation.NonNull;

import org.dhis2afgamis.data.dagger.PerFragment;
import org.dhis2afgamis.data.schedulers.SchedulerProvider;
import org.dhis2afgamis.data.user.UserRepository;
import org.hisp.dhis.android.core.D2;

import dagger.Module;
import dagger.Provides;

/**
 * QUADRAM. Created by ppajuelo on 05/07/2018.
 */
@Module
public class AboutModule {

    @Provides
    @PerFragment
    AboutContracts.AboutPresenter providesPresenter(@NonNull D2 d2, SchedulerProvider provider, @NonNull UserRepository userRepository) {
        return new AboutPresenterImpl(d2, provider, userRepository);
    }
}

File: app/src/main/java/org/medhis2yes/usescases/about/AboutModule.kt
package org.medhis2yes.usescases.about

import dagger.Module
import dagger.Provides
import org.medhis2yes.data.dagger.PerFragment
import org.medhis2yes.data.schedulers.SchedulerProvider
import org.medhis2yes.data.user.UserRepository
import org.hisp.dhis.android.core.D2

@Module
class AboutModule(val view: AboutView) {
    @Provides
    @PerFragment
    fun providesPresenter(
        d2: D2,
        provider: SchedulerProvider,
        userRepository: UserRepository
    ): AboutPresenter {
        return AboutPresenter(view, d2, provider, userRepository)
    }
}
